### ISSUE TITLE
Incorrrect results from timeclock

### ISSUE THREAD

**Issue by anthonyangel** [MindrustUK/Heatmiser-for-home-assistant#2]
Using a neostat in timeclock mode gives some strange results. This issue is really with the Heatmiser API, as the Home Assistant component is just returning the result which the API gives it.

AFAIK the API can return what mode the thermostat is running in, so the irrelevant results could be ignored. 

I'm not sure if the HA climate component allows for switching without temperature, or how this could be implemented. (Potentially would have to be a switch component rather than climate)

![capture](https://cloud.githubusercontent.com/assets/2271981/26462044/22df2a28-4177-11e7-9d58-c8700d44c20e.PNG)

**Comment by MindrustUK** [comment]
Issue necromancy! I think I've got a way around this in terms of changing behaviors. Once I get my second thermostat I'll be revisiting this

**Comment by MindrustUK** [comment]
HASS expecteds a numeric return for temperature. Was going to change this to a "time clock" mode. The device is still a climate type, but maybe needs to be treated in a differnt way removing temperature from the device.

**Comment by leaskovski** [comment]
Just to chime in as I am now using this component and have a unit in the airing cupboard for switching the hot water based on time... if there is no easy solution, then just filtering the device would be fine by me.

**Comment by MindrustUK** [comment]
@leaskovski Thanks for the feedback. Filtering the device is certainly possible, I've got another Neostat on the way to setup as a timeclock for testing. I'll try and add some flags to the plugin to toggle a filtering option in the config. I may also setup some logic to make time clocks appear as switches when I get round to publishing the plugin for that.

**Comment by leaskovski** [comment]
So I got bored this evening and had a look at the JSON coming into HASS.  Here is an example of a thermostat device (underfloor heating) against from timer stat (hot water)...
```
{'AWAY': False, 'COOLING': False, 'COOLING_ENABLED': False, 'COOLING_TEMPERATURE_IN_WHOLE_DEGREES': 0, 'COOL_INP': False, 'COUNT_DOWN_TIME': '0:00', 'CRADLE_PAIRED_TO_REMOTE_SENSOR': False, 'CRADLE_PAIRED_TO_STAT': False, 'CURRENT_FLOOR_TEMPERATURE': 127, 'CURRENT_SET_TEMPERATURE': '21.0', 'CURRENT_TEMPERATURE': '23.2', 'DEMAND': False, 'DEVICE_TYPE': 1, 'ENABLE_BOILER': False, 'ENABLE_COOLING': False, 'ENABLE_PUMP': False, 'ENABLE_VALVE': False, 'ENABLE_ZONE': False, 'FAILSAFE_STATE': False, 'FAIL_SAFE_ENABLED': False, 'FLOOR_LIMIT': False, 'FULL/PARTIAL_LOCK_AVAILABLE': False, 'HEAT/COOL_MODE': False, 'HEATING': False, 'HOLD_TEMPERATURE': 21, 'HOLD_TIME': '0:00', 'HOLIDAY': False, 'HOLIDAY_DAYS': 0, 'HUMIDITY': 0, 'LOCK': False, 'LOCK_PIN_NUMBER': '0000', 'LOW_BATTERY': False, 'MAX_TEMPERATURE': '24.0', 'MIN_TEMPERATURE': '22.0', 'MODULATION_LEVEL': 0, 'NEXT_ON_TIME': '4 days 22:30', 'OFFLINE': False, 'OUPUT_DELAY': False, 'OUTPUT_DELAY': 0, 'PREHEAT': False, 'PREHEAT_TIME': '255:255', 'PROGRAM_MODE': '5DAY/2DAY', 'PUMP_DELAY': False, 'RADIATORS_OR_UNDERFLOOR': False, 'SENSOR_SELECTION': 'BUILT_IN_AIR_SENSOR', 'SET_COUNTDOWN_TIME': 0, 'STANDBY': False, 'STAT_MODE': {'4_HEAT_LEVELS': True, 'MANUAL_OFF': True, 'THERMOSTAT': True}, 'TEMPERATURE_FORMAT': False, 'TEMP_HOLD': False, 'TIMECLOCK_MODE': False, 'TIMER': False, 'TIME_CLOCK_OVERIDE_BIT': False, 'ULTRA_VERSION': 0, 'VERSION_NUMBER': 101, 'WRITE_COUNT': 6, 'ZONE_1PAIRED_TO_MULTILINK': True, 'ZONE_1_OR_2': False, 'ZONE_2_PAIRED_TO_MULTILINK': False, 'device': 'Upstairs'}, 
{'AWAY': False, 'COOLING': False, 'COOLING_ENABLED': False, 'COOLING_TEMPERATURE_IN_WHOLE_DEGREES': 0, 'COOL_INP': False, 'COUNT_DOWN_TIME': '0:00', 'CRADLE_PAIRED_TO_REMOTE_SENSOR': False, 'CRADLE_PAIRED_TO_STAT': False, 'CURRENT_FLOOR_TEMPERATURE': 127, 'CURRENT_SET_TEMPERATURE': '238.0', 'CURRENT_TEMPERATURE': '29.8', 'DEMAND': False, 'DEVICE_TYPE': 1, 'ENABLE_BOILER': False, 'ENABLE_COOLING': False, 'ENABLE_PUMP': False, 'ENABLE_VALVE': False, 'ENABLE_ZONE': False, 'FAILSAFE_STATE': False, 'FAIL_SAFE_ENABLED': False, 'FLOOR_LIMIT': False, 'FULL/PARTIAL_LOCK_AVAILABLE': False, 'HEAT/COOL_MODE': False, 'HEATING': False, 'HOLD_TEMPERATURE': 1, 'HOLD_TIME': '0:00', 'HOLIDAY': False, 'HOLIDAY_DAYS': 0, 'HUMIDITY': 0, 'LOCK': False, 'LOCK_PIN_NUMBER': '0000', 'LOW_BATTERY': False, 'MAX_TEMPERATURE': '255.255', 'MIN_TEMPERATURE': '255.255', 'MODULATION_LEVEL': 0, 'NEXT_ON_TIME': '5 days 05:45', 'OFFLINE': False, 'OUPUT_DELAY': False, 'OUTPUT_DELAY': 0, 'PREHEAT': False, 'PREHEAT_TIME': '255:255', 'PROGRAM_MODE': '5DAY/2DAY', 'PUMP_DELAY': False, 'RADIATORS_OR_UNDERFLOOR': False, 'SENSOR_SELECTION': 'BUILT_IN_AIR_SENSOR', 'SET_COUNTDOWN_TIME': 0, 'STANDBY': False, 'STAT_MODE': {'4_HEAT_LEVELS': True, 'MANUAL_OFF': True, 'TIMECLOCK': True}, 'TEMPERATURE_FORMAT': False, 'TEMP_HOLD': False, 'TIMECLOCK_MODE': False, 'TIMER': False, 'TIME_CLOCK_OVERIDE_BIT': False, 'ULTRA_VERSION': 0, 'VERSION_NUMBER': 101, 'WRITE_COUNT': 23, 'ZONE_1PAIRED_TO_MULTILINK': True, 'ZONE_1_OR_2': False, 'ZONE_2_PAIRED_TO_MULTILINK': False, 'device': 'Hot Water'}]} 
```
The thing I spotted with regards to the difference was for stat mode:
```
'STAT_MODE': {'4_HEAT_LEVELS': True, 'MANUAL_OFF': True, 'THERMOSTAT': True}
'STAT_MODE': {'4_HEAT_LEVELS': True, 'MANUAL_OFF': True, 'TIMECLOCK': True}
```
Filtering these out should be fairly simple, and I might have a go over the next couple of days.

**Comment by leaskovski** [comment]
When I say it should be pretty simple... haha, well that all depends on my lack of python skills.  Anyway, I have added checks into the code in my fork.

**Comment by MindrustUK** [comment]
Added  ExcludeTimeClock = True|False to climate.py I had planned to externalize this to the config but it's giving me the run around. Also supports the 0.96+ of Home Assistant. Still looking for a better way to address the wider issue, it's going to have to sit on the list for a little bit until I get some more time.

**Comment by haakonstorm** [comment]
```
2019-08-09 14:30:08 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for heatmiserneo which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you do experience issues with Home Assistant.
2019-08-09 14:30:09 INFO (MainThread) [homeassistant.components.climate] Setting up climate.heatmiserneo
2019-08-09 14:30:09 INFO (SyncWorker_17) [custom_components.heatmiserneo.climate] Thermostat Name: Oppe
2019-08-09 14:30:09 INFO (SyncWorker_17) [custom_components.heatmiserneo.climate] Thermostat Away Mode: False
2019-08-09 14:30:09 INFO (SyncWorker_17) [custom_components.heatmiserneo.climate] Thermostat Current Temp: 22.4
2019-08-09 14:30:09 INFO (SyncWorker_17) [custom_components.heatmiserneo.climate] Thermostat Set Temp: 21.0
2019-08-09 14:30:09 INFO (SyncWorker_17) [custom_components.heatmiserneo.climate] Thermostat Unit Of Measurement: °C
2019-08-09 14:30:10 INFO (SyncWorker_17) [custom_components.heatmiserneo.climate] Thermostat Name: Gangen
2019-08-09 14:30:10 INFO (SyncWorker_17) [custom_components.heatmiserneo.climate] Thermostat Away Mode: False
2019-08-09 14:30:10 INFO (SyncWorker_17) [custom_components.heatmiserneo.climate] Thermostat Current Temp: 18.7
2019-08-09 14:30:10 INFO (SyncWorker_17) [custom_components.heatmiserneo.climate] Thermostat Set Temp: 25.0
2019-08-09 14:30:10 INFO (SyncWorker_17) [custom_components.heatmiserneo.climate] Thermostat Unit Of Measurement: °C
2019-08-09 14:30:11 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for heatmiserneoplug which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you do experience issues with Home Assistant.
2019-08-09 14:30:14 ERROR (MainThread) [homeassistant.components.climate] Error while setting up platform heatmiserneo
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 126, in _async_setup_platform
    SLOW_SETUP_MAX_WAIT)
  File "/usr/local/lib/python3.7/asyncio/tasks.py", line 442, in wait_for
    return fut.result()
  File "/usr/local/lib/python3.7/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/config/custom_components/heatmiserneo/climate.py", line 91, in setup_platform
    thermostats.append(HeatmiserNeostat(temperature_unit, away, host, port, name))
  File "/config/custom_components/heatmiserneo/climate.py", line 113, in __init__
    self.update()
  File "/config/custom_components/heatmiserneo/climate.py", line 214, in update
    self._hvac_action = STATE_HEAT
NameError: name 'STATE_HEAT' is not defined
2019-08-09 14:30:14 INFO (MainThread) [homeassistant.components.switch] Setting up switch.heatmiserneoplug
2019-08-09 14:30:14 INFO (SyncWorker_3) [custom_components.heatmiserneoplug.switch] Plug Name: neoPlugg, State: on
2019-08-09 14:30:14 INFO (SyncWorker_3) [custom_components.heatmiserneoplug.switch] Adding Switches: [<Entity neoPlugg: off>]
```

<img width="450" alt="Screenshot of Brave Browser Beta (09-08-2019, 14-38-22)" src="https://user-images.githubusercontent.com/193781/62779379-5b964d80-bab3-11e9-8214-2b5d48daad1b.png">

**Comment by haakonstorm** [comment]
FYI: 
```
Update available! 🎉
Home Assistant 0.97.0 is available and you are currently running Home Assistant 0.96.5.
```
I'm on 0.96.5 on HASSIO on RPI3b+.
Upgrading to 0.97.0 to see what happens with the thermostat integration.

The `heatmiserneoplug` still works wonderfully.

**Comment by MindrustUK** [comment]
@haakonstorm let's try and keep the bugs to the right thread. Replying in: https://github.com/MindrustUK/Heatmiser-for-home-assistant/issues/19

**Comment by DrTree** [comment]
This was fixed in #48 I think this issue can be closed

**Comment by MindrustUK** [comment]
As reported by @DrTree. Closed. Thanks.
